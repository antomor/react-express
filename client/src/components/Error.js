import React, {Component} from 'react'

class Error extends Component {
  render(){
    const msg = this.props.msg && this.props.msg !== '' ? 
                this.props.msg :
                '';
    const divMsg = <div>{msg}</div>
    return(
      <div>
        An error occurred:
        {divMsg}
      </div>
    );
  }
}

export default Error;