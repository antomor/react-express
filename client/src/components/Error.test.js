import React from 'react';
import Error from './Error';
import renderer from 'react-test-renderer';

describe('Error', () => {
  it('renders without msg', () => {
    const component = renderer.create(
      <Error />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with msg "Error"', () => {
    const component = renderer.create(
      <Error msg="Error"/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

}); 
