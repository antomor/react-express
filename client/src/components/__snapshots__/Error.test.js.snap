// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders with msg "Error" 1`] = `
<div>
  An error occurred:
  <div>
    Error
  </div>
</div>
`;

exports[`Error renders without msg 1`] = `
<div>
  An error occurred:
  <div>
    
  </div>
</div>
`;
